File: 백준/Silver/1940. 주몽/주몽.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int N = sc.nextInt();
		int M = sc.nextInt();
		int[] nums = new int[N];
		int cnt = 0;
		
		for(int i=0; i<N; i++) {
			nums[i] = sc.nextInt();
		}
		
		Arrays.sort(nums);
		
		int left = 0;
		int right = nums.length-1;
		
		while(left < right) {
			if(nums[left] + nums[right] == M) {
				cnt++;
				left++;
				right--;
			}else {
				if(nums[left] + nums[right] < M) {
					left++;
				}else {
					right--;
				}
			}
		}
		System.out.println(cnt);
	}

}

File: 백준/Silver/1940. 주몽/주몽.kt
import java.util.Arrays

fun main(){
    var N = readln().toInt()
    var M = readln().toInt()
    var nums = readln().split(" ").map{ it.toInt() }.toTypedArray()

    var cnt = 0
    Arrays.sort(nums)

    var start = 0
    var end = nums.size - 1

    while(start < end){
        var sum = nums[start] + nums[end]
        if(sum == M){
            cnt++
            start++
            end--
        }else{
            if(sum > M){
                end--;
            }else{
                start++;
            }
        }
    }
    println(cnt)
}